umbra-js/src/classes/Umbra.ts

```diff
@@ -313,107 +313,161 @@ export class Umbra {
     // that window is defined first to avoid trying to use fetch in node environments
     if (typeof window !== 'undefined' && this.chainConfig.subgraphUrl) {
       try {
-        // Query subgraph
-        const subgraphAnnouncements: SubgraphAnnouncement[] = await recursiveGraphFetch(
-          this.chainConfig.subgraphUrl,
-          'announcementEntities',
-          (filter: string) => `{
-            announcementEntities(${filter}) {
-              amount
-              block
-              ciphertext
-              from
-              id
-              pkx
-              receiver
-              timestamp
-              token
-              txHash
-            }
-          }`
-        );
-
-        // Determine which announcements are for the user.
-        // First we map the subgraph amount field from string to BigNumber, then we reduce the array to the
-        // subset of announcements for the user
-        const announcements = subgraphAnnouncements.map((x) => ({ ...x, amount: BigNumber.from(x.amount) }));
-        const userAnnouncements = announcements.reduce((userAnns, ann) => {
-          const { amount, from, receiver, timestamp, token: tokenAddr, txHash } = ann;
-          const { isForUser, randomNumber } = isAnnouncementForUser(spendingPublicKey, viewingPrivateKey, ann);
-          const token = getAddress(tokenAddr); // ensure checksummed address
-          const isWithdrawn = false; // we always assume not withdrawn and leave it to the caller to check
-          if (isForUser) userAnns.push({ randomNumber, receiver, amount, token, from, txHash, timestamp, isWithdrawn });
-          return userAnns;
-        }, [] as UserAnnouncement[]);
-
-        // Filtering and parsing done, return announcements
-        return { userAnnouncements };
+        // Map the subgraph amount field from string to BigNumber
+        const subgraphAnnouncements = await this.fetchAllAnnouncementsFromSubgraph(startBlock, endBlock);
+        return subgraphAnnouncements.map((x) => ({ ...x, amount: BigNumber.from(x.amount) }));
```

Comment:
comment is off by a line, maybe just inline it
```suggestion
        const subgraphAnnouncements = await this.fetchAllAnnouncementsFromSubgraph(startBlock, endBlock);
        return subgraphAnnouncements.map((x) => ({ ...x, amount: BigNumber.from(x.amount) })); // map subgraph amount field from string to BigNumber
```

Reply:
fixed